include/gsl/span

```diff
@@ -351,16 +356,17 @@ namespace details
 } // namespace details
 
 // [span], class template span
-template <class ElementType, std::ptrdiff_t Extent>
+template <class ElementType, std::size_t Extent>
 class span
 {
 public:
     // constants and types
     using element_type = ElementType;
     using value_type = std::remove_cv_t<ElementType>;
-    using index_type = std::ptrdiff_t;
+    using index_type = std::size_t;
     using pointer = element_type*;
     using reference = element_type&;
+    using difference_type = std::ptrdiff_t;
```

Comment:
Interesting. Based on https://en.cppreference.com/w/cpp/container/span, you do want difference_type as a ptrdiff_t after all. @BillyONeal could you please weigh in here?

Reply:
The `difference_type` of a container is the same as the `difference_type` of its iterators, and the `difference_type` of iterators is always a signed integer type.

Reply:
`difference_type` on a container or view is signed because iterator distances are signed. (Random-access iterators can be subtracted and will return negative if they are transposed.)

Reply:
Thank you Casey and Stephan for your clarification.